packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -144,6 +144,16 @@ abstract class BindingBase {
           };
         }
       );
+      registerServiceExtension(
+          name: 'logging',
+          callback: (Map<String, Object> parameters) async {
+            final String eventKey = parameters['eventKey'];
+            if (eventKey != null) {
```

Comment:
I'd look at `registerBoolServiceExtension` to see how it's implemented; if a bool param is passed, it acts as a toggle, else it just returns the current value. We probably want something similar here.

Reply:
Ah.  Good idea.  I was thinking that channel enablement would be communicated via another, still to come, extension that gives you the state of all channels in one payload but there's no reason not to do this too.  Plus it's probably more generally consistent.  Thanks!

Reply:
If you have a plan already, sgtm. Going forward we probably would want a way to enumerate channels, and to send events when new channels were available.

Reply:
Yeah, I don't think `registerBoolServiceExtension` would work here because you also need the channel name param, but the pattern itself might be useful.